blockchain/filedao/testing.go

```diff
@@ -0,0 +1,227 @@
+package filedao
+
+import (
+	"context"
+	"encoding/hex"
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/iotexproject/go-pkgs/cache"
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/blockchain/block"
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-core/db"
+	"github.com/iotexproject/iotex-core/db/batch"
+	"github.com/iotexproject/iotex-core/test/identityset"
+	"github.com/iotexproject/iotex-core/testutil"
+)
+
+type (
+	// testInMemFd is an in-memory FileDAO
+	testInMemFd struct {
+		*fileDAOv2
+	}
+
+	// testSplitFile will split new db file at certain heights
+	testSplitFile struct {
+		*fileDAO
+		splitHeight []uint64
```

Comment:
you can input a slice []heights to let it split at these heights, used in testing